dimagi/commcare-hq
# Web apps signature capture question type

## Product Description
Makes signature entry questions available in Web Apps using a signature drawing widget.
![image](https://github.com/dimagi/commcare-hq/assets/100609770/0bec3e2a-07c6-4a2b-80b4-305907797a98)


## Technical Summary
[USH-3416](https://dimagi-dev.atlassian.net/browse/USH-3416), [Spec](https://docs.google.com/document/d/1WEpcRt6cUs1uyQjMWUYAT-E_u7PDHjRs1j8QcQoLGgo/edit)
Adds a new javascript dependency, SignaturePad, to allow drawing a signature on an html canvas-based widget. Adds a `SignatureEntry` to display the widget and handle its data. PNG data from the widget is piped into a hidden file input, which gets processed and saved like a regular `FileEntry` question.

## Feature Flag
WEB_APPS_UPLOAD_QUESTIONS

## Safety Assurance

### Safety story
Local testing to make sure signature drawing, clearing and saving works correctly.

### Automated test coverage
Added one test for generating the `SignatureEntry` type, in line with those for other question types.

### QA Plan
[QA-5607](https://dimagi-dev.atlassian.net/browse/QA-5607)
As a question type that is new to web apps, this will want QA to see that behavior is functional and predictable.

### Rollback instructions




### Labels & Review


[USH-3416]: https://dimagi-dev.atlassian.net/browse/USH-3416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[QA-5607]: https://dimagi-dev.atlassian.net/browse/QA-5607?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
